src/Day05.kt

```diff
@@ -33,12 +32,11 @@ fun main() {
     fun part2(input: List<String>): Int {
 
         // Criteria for nice list
-        val pattern = Regex("(..).*\\1")
+        val pattern = Regex("(..).*\\1") 
         fun String.equalEnds() = this.windowed(3).any { it.first() == it.last() }
```

Comment:
The pattern here is:
- (..) matches any two characters as a group
- .* matches any number of any characters (the . stands for any character, while the * indicates that match can be performed indefinitely)
- \\1 matches the same pattern that the most recent group matched. Since there is only one group (..) this will automatically match the first repeated occurrence.